couchdbkit/logging.py

```diff
@@ -30,3 +31,64 @@ def set_logging(level, handler=None):
 
     handler.setFormatter(logging.Formatter(format, datefmt))
     logger.addHandler(handler)
+
+
+def install_request_logger():
+    """Install request logger
+
+    Request metrics are logged to `couchdbkit.request` logger. Extra
+    metrics added to the log record:
+
+    - method
+    - database
+    - path
+    - status_code
+    - content_length
+    - duration
+
+    Returns a function that uninstalls the request logger when called.
+    """
+    from cloudant._client_session import ClientSession
+    from requests.exceptions import HTTPError
+
+    def request(self, method, url, **kwargs):
+        start = default_timer()
+        status_code = None
+        content_length = None
+        try:
+            resp = real_request(self, method, url, **kwargs)
+            status_code = resp.status_code
+            content_length = resp.headers.get("content-length")
+        except HTTPError as err:
+            if err.response is not None:
+                status_code = err.response.status_code
+            raise
+        finally:
+            url_parts = url.split('/', 4)
+            if len(url_parts) == 5:
+                database, path = url_parts[3:]
+            else:
+                database = '<unknown>'
+                path = url
+            info = {
+                "method": method,
+                "database": database,
+                "path": path,
+                "status_code": status_code,
+                "content_length": content_length,
+                "duration": default_timer() - start,
+            }
+            request_logger.debug(
+                '%(method)s to %(database)s/%(path)s took %(duration)s',
+                info,
+                extra=info,
+            )
+        return resp
+
+    def uninstall():
```

Comment:
Thanks for the explanation on this! I think leaving similar comment here would help future me

Reply:
added